app/src/main/java/com/rajatgoyal/puzzle15/ui/MainActivity.java

```diff
@@ -219,14 +227,51 @@ public void setLatestHighScore(HighScore latestHighScore) {
     }
 
     public void init() {
+        SharedPref.init(this);
+
+        resumeBtn = findViewById(R.id.resumeGame);
+        resumeBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(MainActivity.this, GameActivity.class);
+                intent.putExtra("highScoreMoves", latestHighScore.getMoves());
+                intent.putExtra("highScoreTime", latestHighScore.getTime().toSeconds());
+                intent.putExtra("resumeGame", true);
+                startActivity(intent);
+            }
+        });
+
+
         Button newGame = findViewById(R.id.newGame);
         newGame.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Intent i = new Intent(MainActivity.this, GameActivity.class);
-                i.putExtra("highScoreMoves", latestHighScore.getMoves());
-                i.putExtra("highScoreTime", latestHighScore.getTime().toSeconds());
-                startActivity(i);
+                final Intent intent = new Intent(MainActivity.this, GameActivity.class);
+                intent.putExtra("highScoreMoves", latestHighScore.getMoves());
+                intent.putExtra("highScoreTime", latestHighScore.getTime().toSeconds());
+                long prevGameTime = SharedPref.getGameTime();
+                if (prevGameTime > 0) {
+                    String dialogTitle = getResources().getString(R.string.restart_game);
+                    String dialogYesText = getResources().getString(R.string.yes);
```

Comment:
`getResources()` should be stored in a variable rather than calling it a number of times

Reply:
It's cached, but if you say so...